src/decoding/dictionary.rs

```diff
@@ -55,21 +55,15 @@ impl Dictionary {
         let raw_tables = &raw_tables[ll_size as usize..];
 
         let offset1 = &raw_tables[0..4];
-        let offset1 = offset1[0] as u32 + (offset1[1] as u32)
```

Comment:
On a second (third) look, this seems weird. Did you forgot the parentheses or is this really what you wanted? The fix is "correct" because `+` has a higher precedence than `<<` (https://doc.rust-lang.org/reference/expressions.html#expression-precedence) Can you please double check this?

Reply:
Uhm no that looks very wrong. I'll fix this manually. But it should be fixed as soon as I replace all manual little-endian parsing by appropriate functions

Reply:
It seems that there's a testcase missing ;)

Reply:
There are numerous missing, especially for dictionary support. I guess the one I have right now uses a history with [0,0,0], else this should have caused errors. More tests (like more dict tests) and unit tests are the one big thing I will need to do for the next release